internal/dinosaur/pkg/gitops/config.go

```diff
@@ -13,10 +13,58 @@ type Config struct {
 	DataPlaneClusters []DataPlaneClusterConfig `json:"dataPlaneClusters"`
 }
 
+// AuthProviderAddition represents tenant's additional auth provider gitops configuration
+type AuthProviderAddition struct {
+	InstanceID   string        `json:"instanceId"`
+	AuthProvider *AuthProvider `json:"authProvider"`
+}
+
+// AuthProvider represents auth provider configuration
+type AuthProvider struct {
+	Name               string                          `json:"name,omitempty"`
+	MinimumRoleName    string                          `json:"minimumRole,omitempty"`
+	Groups             []AuthProviderGroup             `json:"groups,omitempty"`
+	RequiredAttributes []AuthProviderRequiredAttribute `json:"requiredAttributes,omitempty"`
+	ClaimMappings      []AuthProviderClaimMapping      `json:"claimMappings,omitempty"`
+	OIDCConfig         *AuthProviderOIDCConfig         `json:"oidc,omitempty"`
+}
+
+// AuthProviderRequiredAttribute is representation of storage.AuthProvider_RequiredAttribute that supports transformation from YAML.
+type AuthProviderRequiredAttribute struct {
+	AttributeKey   string `json:"key,omitempty"`
+	AttributeValue string `json:"value,omitempty"`
+}
+
+// AuthProviderClaimMapping represents a single entry in "claim_mappings" field in auth provider proto.
+type AuthProviderClaimMapping struct {
+	Path string `json:"path,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+// AuthProviderGroup is representation of storage.AuthProviderGroup that supports transformation from YAML.
+type AuthProviderGroup struct {
+	AttributeKey   string `json:"key,omitempty"`
+	AttributeValue string `json:"value,omitempty"`
+	RoleName       string `json:"role,omitempty"`
+}
+
+// AuthProviderOIDCConfig contains config values for OIDC auth provider.
+type AuthProviderOIDCConfig struct {
+	Issuer string `json:"issuer,omitempty"`
+	// Depending on callback mode, different OAuth 2.0 would be preferred.
+	// Possible values are: auto, post, query, fragment.
+	CallbackMode string `json:"mode,omitempty"`
+	ClientID     string `json:"clientID,omitempty"`
```

Comment:
Should the json tag be `clientId` rather than `clientID`?
```suggestion
	ClientID     string `json:"clientId,omitempty"`
```